packages/hardhat-ethers/src/helpers.ts

```diff
@@ -148,7 +148,13 @@ async function collectLibrariesAndLink(
   for (const [linkedLibraryName, linkedLibraryAddress] of Object.entries(
     libraries
   )) {
-    if (!utils.isAddress(linkedLibraryAddress)) {
+    // ethers.utils.isAddress may end up being very slow if you pass a complex object to it.
+    // We avoid that by adding this type check.
+    // TODO: add link to issue tracking this.
```

Comment:
No need for a link.

Reply:
Oh, you mean the issue in ethers? I think we won't remove this check even if it's fixed in ethers, so I don't think it's worth it. But I don't mind adding it.

Reply:
Yeah, I don't think it's worth opening an issue in ethers about this, currently.